### PR TITLE
[review] CMake: coding style and best practices

### PR DESCRIPTION
**Pull request by dawagner**
Use known best practices and apply some coding style.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/316%23issuecomment-157374431%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/316%23issuecomment-157674355%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/316%23issuecomment-157715884%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/316%23discussion_r45089163%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/316%23discussion_r45100332%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/316%23discussion_r45199082%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/316%23issuecomment-157374431%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40krocard%20%40tcahuzax%20please%20review%22%2C%20%22created_at%22%3A%20%222015-11-17T13%3A48%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-11-18T10%3A37%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11178583%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tcahuzax%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%20%22%2C%20%22created_at%22%3A%20%222015-11-18T13%3A38%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%203f4e3e2e3e9fcb776425f9e8784edad8e87c2996%20test/functional-tests-legacy/CMakeLists.txt%2011%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/316%23discussion_r45089163%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20strongly%20disapprove%20on%20the%20use%20of%20%60return%28%29%60%20here.%20This%20is%20not%20returning%20from%20the%20current%20block%20but%20rather%20from%20the%20hole%20file.%20Meaning%20that%20an%20subsequent%20include%20would%20not%20be%20evaluated.%20I%20do%20not%20think%20anybody%20expect%20non%20linear%20evaluation%20in%20cmakfiles.%22%2C%20%22created_at%22%3A%20%222015-11-17T17%3A18%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%27m%20scratching%20this%20patch%20anyway%20since%20the%20problem%20it%20tries%20to%20workaround%20is%20gone.%22%2C%20%22created_at%22%3A%20%222015-11-17T18%3A40%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/functional-tests-legacy/CMakeLists.txt%3AL26-75%22%7D%2C%20%22Pull%20eb68c9ed6c5e29b557ba51a086826d628b117578%20xmlserializer/CMakeLists.txt%2025%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/316%23discussion_r45199082%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Fortunately%20all%20this%20push%2C%20pop%20and%20set%20will%20be%20removed%20by%20my%20PR%20%23313.%20%3A%29%20%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-11-18T13%3A38%3A23Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20xmlserializer/CMakeLists.txt%3AL40-59%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/krocard%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/krocard'></a>

- [ ] <a href='#crh-comment-Pull 3f4e3e2e3e9fcb776425f9e8784edad8e87c2996 test/functional-tests-legacy/CMakeLists.txt 11'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/316#discussion_r45089163'>File: test/functional-tests-legacy/CMakeLists.txt:L26-75</a></b>
- <a href='https://github.com/krocard'></a> I strongly disapprove on the use of `return()` here. This is not returning from the current block but rather from the hole file. Meaning that an subsequent include would not be evaluated. I do not think anybody expect non linear evaluation in cmakfiles.
- <a href='https://github.com/dawagner'></a> I'm scratching this patch anyway since the problem it tries to workaround is gone.
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/316#issuecomment-157374431'>General Comment</a></b>
- <a href='https://github.com/dawagner'></a> @krocard @tcahuzax please review
- [x] <a href='#crh-comment-Pull eb68c9ed6c5e29b557ba51a086826d628b117578 xmlserializer/CMakeLists.txt 25'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/316#discussion_r45199082'>File: xmlserializer/CMakeLists.txt:L40-59</a></b>
- <a href='https://github.com/krocard'></a> Fortunately all this push, pop and set will be removed by my PR #313. :) :+1:

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/316?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/316?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/316?approve=0'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/316'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/01org/parameter-framework/pull/316'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by dawagner** [comment]
@krocard @tcahuzax please review

File: test/functional-tests-legacy/CMakeLists.txt [hunk 1]
```diff
@@ -26,41 +26,49 @@
...
 # SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-if (BUILD_TESTING)
+if(BUILD_TESTING)
+    if("${CMAKE_CURRENT_BINARY_DIR}" MATCHES ".* .*")
+        message(WARNING
+            "The build directory path contains a space. Unfortunately, the"
+            " legacy functional tests are known not to work in such a case."
+            " For this reason, they will be deactivated for this build.")
+        return()
```
Comment (krocard) [inline test/functional-tests-legacy/CMakeLists.txt]: I strongly disapprove on the use of `return()` here. This is not returning from the current block but rather from the hole file. Meaning that an subsequent include would not be evaluated. I do not think anybody expect non linear evaluation in cmakfiles.

Comment (dawagner) [inline test/functional-tests-legacy/CMakeLists.txt]: I'm scratching this patch anyway since the problem it tries to workaround is gone.

**Comment by tcahuzax** [comment]
:+1:

File: xmlserializer/CMakeLists.txt [hunk 2]
```diff
@@ -40,15 +40,19 @@ set_target_properties(xmlserializer PROPERTIES POSITION_INDEPENDENT_CODE TRUE)
...
+cmake_reset_check_state()
+include(CheckSymbolExists)
+set(CMAKE_REQUIRED_DEFINITIONS ${LIBXML2_DEFINITIONS})
+set(CMAKE_REQUIRED_INCLUDES ${LIBXML2_INCLUDE_DIR})
+# Check that the LIBXML_XINCLUDE_ENABLED macro is defined
+# This is the safest check, as it does not involve linking against anything
+# (which might fail for another reason than the lack of XInclude support)
+check_symbol_exists(LIBXML_XINCLUDE_ENABLED libxml/xmlversion.h XML2_XINCLUDE_SUPPORT)
+# Restore the previous state
+cmake_pop_check_state()
```
Comment (krocard) [inline xmlserializer/CMakeLists.txt]: Fortunately all this push, pop and set will be removed by my PR #313. :) :+1:

**Comment by krocard** [comment]
:shipit:

**Pull request merged**